target/linux/lantiq/files/arch/mips/boot/dts/lantiq/vr9_avm_fritz7490.dts

```diff
@@ -0,0 +1,285 @@
+/dts-v1/;
+
+#include "vr9.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/mips/lantiq_rcu_gphy.h>
+
+/ {
+	compatible = "avm,fritz7490", "lantiq,xway", "lantiq,vr9";
+	model = "AVM FRITZ!Box 7490";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200";
+	};
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_info_red;
+		led-running = &led_power;
+		led-upgrade = &led_info_red;
+
+		led-dsl = &led_internet;
+		led-internet = &led_internet;
+		led-wifi = &led_wlan;
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x10000000>;
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
+
+		dect {
+			label = "dect";
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_PHONE>;
+		};
+
+		wlan {
+			label = "wlan";
+			gpios = <&gpio 29 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RFKILL>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "fritz7490:green:power";
+			gpios = <&gpio 45 GPIO_ACTIVE_LOW>;
+			default-state = "keep";
+		};
+
+		led_internet: internet {
+			label = "fritz7490:green:internet";
+			gpios = <&gpio 47 GPIO_ACTIVE_LOW>;
+		};
+
+		led_fixedline {
+			label = "fritz7490:green:fixedline";
+			gpios = <&gpio 36 GPIO_ACTIVE_LOW>;
+		};
+
+		led_wlan: wlan {
+			label = "fritz7490:green:wlan";
+			gpios = <&gpio 35 GPIO_ACTIVE_LOW>;
+		};
+
+		led_info_green {
+			label = "fritz7490:green:info";
+			gpios = <&gpio 33 GPIO_ACTIVE_LOW>;
+		};
+
+		led_info_red: info_red {
+			label = "fritz7490:red:info";
+			gpios = <&gpio 46 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+
+		gpio_wasp_reset {
+			gpio-export,name = "fritz7490:wasp:reset";
+			gpio-export,output = <1>;
+			gpios = <&gpio 34 GPIO_ACTIVE_LOW>;
+		};
+
+		gpio_wasp_wakeup {
+			gpio-export,name = "fritz7490:wasp:wakeup";
+			gpio-export,output = <2>;
+			gpios = <&gpio 5 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&eth0 {
+	interface@0 {
+		compatible = "lantiq,xrx200-pdi";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0>;
+		lantiq,switch;
+
+		ethernet@0 {
+			compatible = "lantiq,xrx200-pdi-port";
+			reg = <0>;
+			phy-mode = "sgmii";
+			phy-handle = <&phy0>;
+			gpios = <&gpio 32 GPIO_ACTIVE_HIGH>;
+		};
+
+		ethernet@1 {
+			compatible = "lantiq,xrx200-pdi-port";
+			reg = <1>;
+			phy-mode = "sgmii";
+			phy-handle = <&phy1>;
+			gpios = <&gpio 44 GPIO_ACTIVE_HIGH>;
+		};
+
+		ethernet@2 {
+			compatible = "lantiq,xrx200-pdi-port";
+			reg = <2>;
+			phy-mode = "gmii";
+			phy-handle = <&phy11>;
+		};
+
+		ethernet@4 {
+			compatible = "lantiq,xrx200-pdi-port";
+			reg = <4>;
+			phy-mode = "gmii";
+			phy-handle = <&phy13>;
+		};
+
+		ethernet@5 {
+			compatible = "lantiq,xrx200-pdi-port";
+			reg = <5>;
+			phy-mode = "rgmii";
+
+			fixed-link {
```

Comment:
The xrx200 driver does not support fixed links unless you add a patch.

Reply:
yes, do you mean this one?:

https://github.com/andyboeh/openwrt/commit/9eeecc79da5940b54344995dfa648fe944772174#diff-53548cba5fb20e52886933f9e46e869e

Then the question needs tobe resolved how do we register this port in the userspace domain, I can imagine we need to label it as "wan" but I am not sure of the implications...

ucidef_add_switch "switch0" \
		"0:lan:3" "1:lan:4" "2:lan:2" "4:lan:1" "5:wan:5" "6t@eth0"

Reply:
Hmmm and.. possibly they hooked up the Xilinx to port 3 maybe.

If this is true this could opens up some interesting possibilities... an onboard programmable  FPGA on the network ... :)

Reply:
Yes, I meant this commit. Without it, the lines regarding `fixed-link` are useless.

There is an ongoing discussion on the labeling of this port in the 3390 PR. My point of view is that it should bear a "lan" label, because the WiFi Access Point is connected to it.

Reply:
I would agree. (I was under the wrong impression that fixed link meant the DSL soc...)
ucidef_add_switch "switch0" \
           "0:lan:3" "1:lan:4" "2:lan:2" "4:lan:1" "5:lan:5" "6t@eth0"

Reply:
No, it means that the kernel fakes a PHY and an MDIO bus at the given port with the given speed / settings.

Reply:
please remove this till it is really supported.